packages/askar/src/AskarModule.ts

```diff
@@ -1,33 +1,18 @@
+import type { AskarModuleConfigOptions } from './AskarModuleConfig'
 import type { DependencyManager, Module } from '@aries-framework/core'
 
-import { AriesFrameworkError, InjectionSymbols } from '@aries-framework/core'
-
+import { AskarModuleConfig } from './AskarModuleConfig'
 import { AskarStorageService } from './storage'
-import { AskarWallet } from './wallet'
 
 export class AskarModule implements Module {
-  public register(dependencyManager: DependencyManager) {
-    try {
-      // eslint-disable-next-line import/no-extraneous-dependencies
-      require('@hyperledger/aries-askar-nodejs')
-    } catch (error) {
-      try {
-        require('@hyperledger/aries-askar-react-native')
-      } catch (error) {
-        throw new Error('Could not load aries-askar bindings')
-      }
-    }
+  public readonly config: AskarModuleConfig
 
-    if (dependencyManager.isRegistered(InjectionSymbols.Wallet)) {
-      throw new AriesFrameworkError('There is an instance of Wallet already registered')
-    } else {
-      dependencyManager.registerContextScoped(InjectionSymbols.Wallet, AskarWallet)
-    }
+  public constructor(config: AskarModuleConfigOptions) {
+    this.config = new AskarModuleConfig(config)
+  }
+  public register(dependencyManager: DependencyManager) {
+    dependencyManager.registerInstance(AskarModuleConfig, this.config)
 
-    if (dependencyManager.isRegistered(InjectionSymbols.StorageService)) {
-      throw new AriesFrameworkError('There is an instance of StorageService already registered')
-    } else {
-      dependencyManager.registerSingleton(InjectionSymbols.StorageService, AskarStorageService)
-    }
+    dependencyManager.registerSingleton(AskarStorageService)
```

Comment:
Would be good to keep this check

Reply:
@TimoGlastra  keep this check along with my new implementation?
Ex.
```ts
  public register(dependencyManager: DependencyManager) {
    dependencyManager.registerInstance(AskarModuleConfig, this.config)

    dependencyManager.registerSingleton(AskarStorageService)

    if (dependencyManager.isRegistered(InjectionSymbols.Wallet)) {
      throw new AriesFrameworkError('There is an instance of Wallet already registered')
    } else {
      dependencyManager.registerContextScoped(InjectionSymbols.Wallet, AskarWallet)
    }

    if (dependencyManager.isRegistered(InjectionSymbols.StorageService)) {
      throw new AriesFrameworkError('There is an instance of StorageService already registered')
    } else {
      dependencyManager.registerSingleton(InjectionSymbols.StorageService, AskarStorageService)
    }
  }
```

Reply:
In this particular case is good to keep it because if there is already a Wallet and a StorageService registered (which means that there is other module serving to the same functionality) and we simply register the symbols, it will override the classes registered by the other module, which is for sure not a desired behaviour. So this will force the user to 'do the right thing' and not use two equivalent modules (it's like using all these new components and indy-sdk at the same time).